untrusted/urts/src/lib.rs

```diff
@@ -59,6 +118,21 @@ impl EnclaveBuilder {
         self
     }
 
+    /// Report function to use for the built [Enclave].
+    ///
+    /// This will be used when calling [Enclave.report()]
+    ///
+    /// # Arguments
+    ///
+    /// * `report_fn` - The function to use for creating [Enclave] reports,
+    ///     [Enclave::report()].  This is `None` by default.  Passing in will
+    ///     prevent creating reports, [Error::NoReportFunction].
+    #[must_use]
+    pub fn report_fn(mut self, report_fn: Option<EnclaveReportFn>) -> EnclaveBuilder {
```

Comment:
When would we not want a `ReportFn`? This might be better as a trait, like [`ReportableEnclave`](https://github.com/mobilecoinfoundation/mobilecoin/blob/5a0be430bece7d52495737746b42e65e5573ee6b/sgx/report-cache/api/src/lib.rs#L59)

Reply:
> When would we not want a `ReportFn`?
That's a good question. I think for running production/release enclaves probably almost never.
Getting an enclave up and going even for debug purposes has quite a few boxes to check so I made this optional to lower the barrier to entry a little bit.  I'll admit it may be complicating and over engineering. 

As for the trait, I think it sounds like a good idea, and a good reminder I need to look at them more, but I'm not quite seeing how to do it.  

Defining the trait in urts/src/lib.rs prevents one from implementing it on [Enclave].
Defining the trait in the in progress DCAP quote generation crate requires that crate to know what the "enclave.so" function is.  

The main issue is that the "enclave.so" specific function has to be called and it can be specific to the enclave which means one has to provide the enclave specific function somehow to these more generic interfaces.

Reply:
I was thinking we'd have a `ReportableEnclave` trait, but then we'd need a `ReportableEnclaveBuilder`, too. Since we're not currently defining such traits, leaving it as an `Option<EnclaveReportFn>` SGTM